### PR TITLE
Add 'Reset' option to MediaReplaceFlow component

### PR DESCRIPTION
**Pull request by akasunil**
## What? and Why?

`Reset` option is used to clear media from block, such as image block, cover block etc. However, the `MediaReplaceFlow` component does not by default have the `Reset` option available. We used to add the `Reset` option as a child of the `MediaReplaceFlow` component, Whenever necessary. We had to include the `Reset` option in the `MediaReplaceFlow` component itself because it is a fundamental feature that is needed anywhere media is used, saving us from having to write additional code.
 
## How?

This PR,
- Adds `Reset` option to `MediaReplaceFlow` Component.
- Update implementation of `MediaReplaceFlow` component in `Image`, `Cover`, `Post Featured Image`, `Site Logo` blocks
- Adds `Reset` option to `Audio`, `Video`, `File` and `Media-Text` block.
- Also, Update implementation of `MediaReplaceFlow` component in background support under global styles.

## Testing Instructions
- Edit post
- Add block from any of ( `Image`, `Cover`, `Post Featured Image`, `Site Logo` , `Audio`, `Video`, `File`, `Media-Text` )
- Select media into added block
- Ensure `Reset` option is visible in toolbar
- Ensure media is removed on click of `Reset`

## Screenshots or screencast 

| Block  | Reset option demo |
| ------------- | ------------- |
| Image  |  image-block |
| Cover  | cover-block  |
| Post Featured Image  | featured-image-block  |
| Site Logo  | site-logo  |
| Audio  | audio-block  |
| Video  | video-block  |
| File  | file-block  |
| Media Text  | media-text-block |

### CODE REVIEW

**Comment by jasmussen** [comment]
Nice, this feels like a good cleanup and optimization. I think this just needs a code review. Great PR!

File: packages/block-library/src/media-text/media-container.js [hunk 1]
```diff
@@ -62,22 +62,18 @@ function ToolbarEditButton( {
...
 } ) {
 	return (
 		<BlockControls group="other">
 			<MediaReplaceFlow
 				mediaId={ mediaId }
-				mediaUrl={
-					useFeaturedImage && featuredImageURL
-						? featuredImageURL
-						: mediaUrl
-				}
```
Comment (Mamaduka) [inline packages/block-library/src/media-text/media-container.js:75]: This looks unrelated. Can you explain why we're removing special handling for featured images?

File: packages/block-editor/src/components/global-styles/background-panel.js [hunk 2]
```diff
@@ -432,6 +432,10 @@ function BackgroundImageControls( {
 				}
 				variant="secondary"
 				onError={ onUploadError }
+				onReset={ () => {
+					closeAndFocus();
+					onResetImage();
```
Comment (Mamaduka) [inline packages/block-editor/src/components/global-styles/background-panel.js:437]: Besides this component and the Cover block, all reset callbacks are the same as `onSelect( undefined )`. I wonder if we really need a new `onReset` callback.

File: packages/block-library/src/media-text/media-container.js (hunk 1 above)
Comment (akasunil) [reply packages/block-library/src/media-text/media-container.js:75]: This condition is already handled [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/media-text/media-container.js#L196-L200) when passing argument to `ToolbarEditButton`. Don't see any need to add same condition again in `ToolbarEditButton` component.

Comment (Mamaduka) [reply packages/block-library/src/media-text/media-container.js:75]: Makes sense!

File: packages/block-editor/src/components/global-styles/background-panel.js (hunk 2 above)
Comment (akasunil) [reply packages/block-editor/src/components/global-styles/background-panel.js:437]: We can make it optional by setting `onSelect( undefined )` as default action for `Reset` option.

Comment (Mamaduka) [reply packages/block-editor/src/components/global-styles/background-panel.js:437]: That adds more logic that `MediaReplaceFlow` needs to handle, and I was thinking the opposite; it already supports too many  "on-action" callback props.

@ramonjd, @andrewserong, what do you think?

Comment (ramonjd) [reply packages/block-editor/src/components/global-styles/background-panel.js:437]: I'm torn.

A benefit is that consumers don't have to include their own `<MenuItem />` and `onClose()` functionality. I see that, in trunk, the menu hangs after reset

Screenshot 2024-08-29 at 12 25 42 PM

The downside is that the "Remove" item won't be customizable for each block or file type. What if, later, the copy needs to update, e.g., `Remove [specific item]` or extra accessibility props — then we'd be loading `<MediaReplaceFlow />` again. For that reason this optimization could be premature.

Comment (ramonjd) [reply packages/block-editor/src/components/global-styles/background-panel.js:437]: > and onClose() functionality

Child functions get passed that any way I see:

https://github.com/WordPress/gutenberg/blob/d82ce72ffc9260d3ef1693fbdc9c28ca7d7bccbc/packages/block-editor/src/components/media-replace-flow/index.js#L206-L206

Comment (Mamaduka) [reply packages/block-editor/src/components/global-styles/background-panel.js:437]: After adding "Reset" support to most media blocks in the core, I see this as more consolidation than optimization.

I'm pro-composition, but here, it seems we want to have the same feature/label across the blocks.

Comment (andrewserong) [reply packages/block-editor/src/components/global-styles/background-panel.js:437]: Good discussion! I suppose one potential challenge with removing the `onRemove` prop and consolidating around it calling `onSelect( undefined )` is that the `MediaReplaceFlow` component appears to be a public / stable one. Would consumers be guaranteed to handle it, i.e. what if someone has a callback that always expects a value? And should consumers always have a Reset option?

I suppose one of the neat things about the `onRemove` prop is that if consumers don't provide a callback, the option will never be displayed, so it's backwards compatible in case anyone has code that wouldn't handle `onSelect( undefined )` gracefully.

But I don't feel too strongly either way on this topic 🙂

Comment (Mamaduka) [reply packages/block-editor/src/components/global-styles/background-panel.js:437]: Thanks, good point, @andrewserong!

If consumers don't expect the "Reset" button, the block editor shouldn't surprise them with one :)

Having a new `onReset` prop and conditional rendering will allow us that.

**Review by Mamaduka — APPROVED** [review]

**Comment by akasunil** [comment]
Thank you @Mamaduka, @ramonjd and @andrewserong. I'll Merge this PR now.
